gradoop-temporal/src/test/java/org/gradoop/temporal/util/TemporalGradoopTestBase.java

```diff
@@ -153,6 +156,48 @@ protected TemporalGraphCollection toTemporalGraphCollection(BaseGraphCollection<
     return getConfig().getTemporalGraphCollectionFactory().fromNonTemporalGraphCollection(collection);
   }
 
+  /**
+   * Convert a graph collection to a {@link TemporalGraphCollection} and set temporal attributes using
+   * {@link TimeIntervalExtractor} functions.
+   *
+   * @param collection             The graph collection.
+   * @param graphHeadTimeExtractor The function used to extract temporal attributes for graph heads.
+   * @param vertexTimeExtractor    The function used to extract temporal attributes for vertices.
+   * @param edgeTimeExtractor      The function used to extract temporal attributes for edges.
+   * @param <G> The graph head type.
+   * @param <V> The vertex type.
+   * @param <E> The edge type.
+   * @return A temporal graph with temporal attributes extracted from the original graph.
+   */
+  protected <G extends GraphHead, V extends Vertex, E extends Edge> TemporalGraphCollection
+  toTemporalGraphCollection(
+    BaseGraphCollection<G, V, E, ?, ?> collection,
+    TimeIntervalExtractor<G> graphHeadTimeExtractor,
+    TimeIntervalExtractor<V> vertexTimeExtractor,
+    TimeIntervalExtractor<E> edgeTimeExtractor) {
+    return getConfig().getTemporalGraphCollectionFactory().fromNonTemporalDataSets(
+      collection.getGraphHeads(), graphHeadTimeExtractor, collection.getVertices(), vertexTimeExtractor,
+      collection.getEdges(), edgeTimeExtractor);
+  }
+
+  /**
+   * Convert a graph collection to a {@link TemporalGraphCollection} with time extraction functions.
+   * This will use {@link TemporalGradoopTestUtils#extractTime(Element)} to extract temporal attributes.
+   *
+   * @param collection The graph collection.
+   * @return The temporal graph with extracted temporal information.
+   */
+  protected TemporalGraphCollection toTemporalGraphCollectionWithDefaultExtractors(
+    BaseGraphCollection<?, ?, ?, ?, ?> collection) {
+    // We have to use lambda expressions instead of method references here, otherwise a
+    // ClassCastException will be thrown when those extractor functions are called.
+    // TODO: Find out why.
```

Comment:
Imo its better to open an issue with that request. TODO tend to just be forgotten?